### PR TITLE
User session

### PR DESCRIPTION
**Pull request by mxaly**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,35 @@
+class SerializserScope
```
Comment (nekath) [inline app/controllers/application_controller.rb]: typo

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    :authorization_uri => $credentials.authorization_uri,
+    :token_credential_uri => $credentials.token_credential_uri,
+    :client_id => $credentials.client_id,
+    :client_secret => $credentials.client_secret,
+    :redirect_uri => $credentials.redirect_uris.first,
+    :access_type => 'offline',
+    :scope => PLUS_LOGIN_SCOPE)
+$client = Google::APIClient.new
+
+$allowed_domians = %w(gmail.com netguru.co netguru.pl)
```
Comment (nekath) [inline app/controllers/sessions_controller.rb]: move it to config!

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class User < ActiveRecord::Base
+  def generate_token!
+    update_attribute :token, SecureRandom.hex(8)
+  rescue ActiveRecord::RecordNotUnique
+    retry
+  end
```
Comment (nekath) [inline app/models/user.rb]:
```ruby
def add_token!
  new_token = loop do
     safe_token = SecureRandom.hex(8)
     break safe_token unless User.where(token: token).exists?
  end

  update_attributes(token: new_token)
end
```

Comment (mxaly) [inline app/models/user.rb]: this way, no metter if token is a duplicate or not, there will be extra query `User.where(token: token).exists?`

**Pull request merged**